### PR TITLE
Update changelog with SwerveDriveOdometry changes, remove incorrect info

### LINKED ISSUE #1966: Document that mecanum/swerve odometry and pose estimator classes replaced speeds with positions in 2023 changelog
From https://github.com/wpilibsuite/BetaTest/discussions/14#discussioncomment-4103776:
> * SwerveDriveOdometry requires SwerveModulePositions instead of SwerveModuleState, and the distinction between the two wasn't immediately clear, or even whether the Positions class was intended as a replacement or not.

### PR DESCRIPTION
**Pull request by rzblue**
Closes #1966

`CameraServer` was not removed

### CODE REVIEW

File: source/docs/yearly-overview/yearly-changelog.rst [hunk 1]
```diff
@@ -88,7 +88,6 @@ Breaking Changes
 .. danger:: Updated ``DifferentialDrive`` and ``MecanumDrive`` classes to use North-West-Up axis conventions to match the rest of WPILib. The Z-axis (i.e. turning) will need to be inverted to restore the old behavior.
 
 - Removed deprecated ``MakeMatrix()`` from ``StateSpaceUtil``
-- Removed deprecated ``CameraServer`` class
```
Comment (sciencewhiz) [inline source/docs/yearly-overview/yearly-changelog.rst:91]: It was the getInstance method of the CameraServer class, not the whole class, but should still be mentioned.

https://github.com/wpilibsuite/allwpilib/pull/4205

**Review by sciencewhiz — CHANGES_REQUESTED** [review]
It was deprecated getInstance method that wa

File: source/docs/yearly-overview/yearly-changelog.rst (hunk 1 above)
Comment (rzblue) [reply source/docs/yearly-overview/yearly-changelog.rst:91]: `getInstance` methods are covered later down in the list

**Review by sciencewhiz — APPROVED** [review]

**Comment by rzblue** [comment]
Rebased to resolve conflict

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**